docs/linux/setup_linux-host_qemu-vm_riscv64-kernel.md

```diff
@@ -0,0 +1,153 @@
+# Setup: Debian/Ubuntu host, QEMU vm, riscv64 kernel
+
+# GCC
+
+Obtain `riscv64-linux-gnu-gcc` and `riscv-unknown-elf-gcc`, at least GCC version 8. The latest
+Debian/Ubuntu distributions should provide both cross-compilers in a recent enough version.
```

Comment:
Please also provide required package names.
I found these: `gcc-riscv64-linux-gnu gcc-riscv64-unknown-elf`.

Unfortunately it does not work on my Debian testing fork:
```
$ riscv64-unknown-elf-gcc /tmp/test.c
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find crt0.o: No such file or directory
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lc
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lgloss
collect2: error: ld returned 1 exit status
```
Is there anything else I need to install, or it's just broken packages in my distro?

Reply:
> Please also provide required package names.
> I found these: `gcc-riscv64-linux-gnu gcc-riscv64-unknown-elf`.

Yeah, this are the correct ones. Will add them to the doc.

> Unfortunately it does not work on my Debian testing fork:
> 
> ```
> $ riscv64-unknown-elf-gcc /tmp/test.c
> /usr/lib/riscv64-unknown-elf/bin/ld: cannot find crt0.o: No such file or directory
> /usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lc
> /usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lgloss
> collect2: error: ld returned 1 exit status
> ```
> 
> Is there anything else I need to install, or it's just broken packages in my distro?

This looks the same on my Ubuntu system and with the manually built toolchain.

It seems `riscv64-unknown-elf-gcc` will not work without additional flags In the scope of this doc it is only needed to build BBL (see the corresponding section below in the doc), which provides all the necessary flags to it. This worked fine with the binary from the `riscv64-unknown-elf-gcc 8.3.0.2019.08+dfsg-1` package

Reply:
I might be wrong about `riscv64-unknown-elf-gcc 8.3.0.2019.08+dfsg-1` working. Looks like I had a manually build toolchain in my `$PATH` :disappointed:

Reply:
I did this compilation test because:
```
build$ ../configure --enable-logo --host=riscv64-unknown-elf --with-payload=/usr/local/google/home/dvyukov/src/linux2/vmlinux
checking build system type... x86_64-pc-linux-gnu
checking host system type... riscv64-unknown-elf
checking for riscv64-unknown-elf-gcc... riscv64-unknown-elf-gcc
checking whether the C compiler works... no
configure: error: in `/usr/local/google/home/dvyukov/src/riscv-pk/build':
configure: error: C compiler cannot create executables
See `config.log' for more details

build$ cat config.log
...
configure:2382: checking whether the C compiler works
configure:2404: riscv64-unknown-elf-gcc    conftest.c  >&5
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find crt0.o: No such file or directory
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lc
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lgloss
collect2: error: ld returned 1 exit status
...
```

Reply:
It seems `--host=riscv64-linux-gnu` works fine as per your comment https://github.com/google/syzkaller/pull/1875#discussion_r446506423, so we can drop `riscv64-unknown-elf-gcc` altogether. Sorry for the confusion.